src/gui/programupdater.cpp

```diff
@@ -36,66 +37,95 @@
 #include <QDebug>
 #include <QDesktopServices>
 #include <QRegularExpression>
-#include <QStringList>
 #include <QXmlStreamReader>
 
 #if defined(Q_OS_WIN)
 #include <QSysInfo>
 #endif
 
 #include "base/net/downloadmanager.h"
+#include "base/utils/version.h"
 #include "base/version.h"
 
 namespace
 {
-    const QString RSS_URL {QStringLiteral("https://www.fosshub.com/feed/5b8793a7f9ee5a5c3e97a3b2.xml")};
+    bool isVersionMoreRecent(const QString &remoteVersion)
+    {
+        using Version = Utils::Version<int, 4, 3>;
 
-    QString getStringValue(QXmlStreamReader &xml);
+        try
+        {
+            const Version newVersion {remoteVersion};
+            const Version currentVersion {QBT_VERSION_MAJOR, QBT_VERSION_MINOR, QBT_VERSION_BUGFIX, QBT_VERSION_BUILD};
+            if (newVersion == currentVersion)
+            {
+                const bool isDevVersion = QString::fromLatin1(QBT_VERSION_STATUS).contains(
+                    QRegularExpression(QLatin1String("(alpha|beta|rc)")));
+                if (isDevVersion)
+                    return true;
+            }
+            return (newVersion > currentVersion);
+        }
+        catch (const std::runtime_error &)
+        {
+            return false;
+        }
+    }
 }
 
-ProgramUpdater::ProgramUpdater(QObject *parent, bool invokedByUser)
-    : QObject(parent)
-    , m_invokedByUser(invokedByUser)
+ProgramUpdater::ProgramUpdater(QObject *parent)
```

Comment:
Now you can use `using QObject::QObject;` in header instead.

Reply:
Done.